### PR TITLE
DRILL-5258: Access mock data definition from SQL

### PR DESCRIPTION
**Pull request by paul-rogers**
Extends the mock data source to allow using the full power of the mock
data source from an SQL query by referencing the JSON definition
file. See JIRA and package-info for details.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/mock/MockGroupScanPOP.java [hunk 1]
```diff
@@ -75,20 +76,50 @@
...
+          if (col.getWidthValue() == 0) {
+            colWidth = TypeHelper.getSize(col.getMajorType());
+          } else {
+            colWidth = col.getWidthValue();
+          }
+          colWidth *= col.getRepeatCount();
+          width += colWidth;
+        }
+      }
+      rowWidth = Math.max(rowWidth, width);
```
Comment (sohami) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/mock/MockGroupScanPOP.java]: `rowWidth` seems to be `maxRowWidth` and `width` is `rowWidth`. Can we please rename these ?

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/mock/MockStorageEngine.java [hunk 2]
```diff
@@ -109,7 +121,37 @@ public MockSchema(MockStorageEngine engine) {
...
+        mapper.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true);
+        mockTableDefn = mapper.readValue(json, MockTableDef.class);
+      } catch (JsonParseException e) {
+        throw new IllegalArgumentException( "Unable to parse mock table definition file: " + name, e );
+      } catch (JsonMappingException e) {
+        throw new IllegalArgumentException( "Unable to Jackson deserialize mock table definition file: " + name, e );
+      } catch (IOException e) {
+        throw new IllegalArgumentException( "Unable to read mock table definition file: " + name, e );
+      }
+      return new DynamicDrillTable(engine, this.name, mockTableDefn.getEntries() );
```
Comment (sohami) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/mock/MockStorageEngine.java]: Please remove extra space before last` )`. there are few other places too like line 169 in this file, line 199 (MockTableDef.java), etc.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/mock/package-info.java [hunk 3]
```diff
@@ -60,14 +62,26 @@
  * <li>The <tt>mode</tt> is one of the supported Drill
  * {@link DataMode} names: usually <tt>OPTIONAL</tt> or <tt>REQUIRED</tt>.</li>
  * </ul>
+ * <p>
+ * Recent extensions include:
+ * <ul>
+ * <li><tt>repeat</tt> in either the "entry" or "record" elements allow
```
Comment (paul-rogers) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/mock/package-info.java]: Yes. Added the property to MockScanEntry. Need to add it to the implementation as well, which is planned, but not yet complete.

File: exec/java-exec/src/test/java/org/apache/drill/exec/fn/interp/ExpressionInterpreterTest.java [hunk 4]
```diff
@@ -124,7 +125,7 @@ public void interpreterDateTest() throws Exception {
     final BitControl.PlanFragment planFragment = BitControl.PlanFragment.getDefaultInstance();
     final QueryContextInformation queryContextInfo = planFragment.getContext();
     final int                        timeZoneIndex = queryContextInfo.getTimeZone();
-    final org.joda.time.DateTimeZone timeZone = org.joda.time.DateTimeZone.forID(org.apache.drill.exec.expr.fn.impl.DateUtility.getTimeZone(timeZoneIndex));
+    final DateTimeZone timeZone =    DateTimeZone.forID(org.apache.drill.exec.expr.fn.impl.DateUtility.getTimeZone(timeZoneIndex));
```
Comment (paul-rogers) [inline exec/java-exec/src/test/java/org/apache/drill/exec/fn/interp/ExpressionInterpreterTest.java]: Original code, but fixed.

**Comment by sohami** [comment]
Thanks for the change. LGTM. +1

**Pull request closed without merging**